lib/models/os.ts

```diff
@@ -382,27 +372,26 @@ const getOsModel = function (
 	 * 	console.log('The raspberry-pi image was last modified in ' + date);
 	 * });
 	 */
-	const getLastModified = function (
+	const getLastModified = async function (
 		deviceType: string,
 		version: string = 'latest',
 	): Promise<Date> {
-		return getValidatedDeviceType(deviceType)
-			.then(() => normalizeVersion(version))
-			.then((ver) =>
-				request.send({
-					method: 'HEAD',
-					url: deviceImageUrl(deviceType, ver),
-					baseUrl: apiUrl,
-					// optionally authenticated, so we send the token in all cases
-				}),
-			)
-			.catch((err) => {
-				if (isNotFoundResponse(err)) {
-					throw new Error('No such version for the device type');
-				}
-				throw err;
-			})
-			.then((response) => new Date(response.headers.get('last-modified')!));
+		let response;
+		try {
+			await getValidatedDeviceType(deviceType);
+			const ver = normalizeVersion(version);
+			response = await request.send({
+				method: 'HEAD',
+				url: deviceImageUrl(deviceType, ver),
+				baseUrl: apiUrl,
+			});
+		} catch (err) {
+			if (isNotFoundResponse(err)) {
+				throw new Error('No such version for the device type');
+			}
+			throw err;
+		}
+		return new Date(response.headers.get('last-modified')!);
```

Comment:
Can be moved inside the `try` and we can then define `const response` during the assignment.